api/src/main/java/io/opencensus/trace/CallableSpanHandle.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2017, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opencensus.trace;
+
+import com.google.common.base.Throwables;
+import java.util.concurrent.Callable;
+
+final class CallableSpanHandle<V> implements Callable<V> {
+  private final Span span;
+  private final Callable<V> callable;
+
+  CallableSpanHandle(Span span, Callable<V> callable) {
+    this.span = span;
+    this.callable = callable;
+  }
+
+  @Override
+  public V call() throws Exception {
+    try {
+      return callable.call();
+    } catch (Exception e) {
+      span.setStatus(Status.UNKNOWN.withDescription(e.getMessage()));
+      throw e;
+    } catch (Throwable t) {
+      span.setStatus(Status.UNKNOWN.withDescription(t.getMessage()));
```

Comment:
guard against getMessage() == null. Ex I tend to use getClass().getSimpleName when message is null.